lxd/device/gpu_mdev.go

```diff
@@ -127,6 +131,17 @@ func (d *gpuMdev) startVM() (*deviceConfig.RunConfig, error) {
 
 				return nil, errors.Wrapf(err, "Failed to create virtual gpu %q", mdevUUID)
 			}
+
+			revert.Add(func() {
+				path := fmt.Sprintf("/sys/bus/mdev/devices/%s", mdevUUID)
+
+				if shared.PathExists(path) {
+					err := ioutil.WriteFile(filepath.Join(path, "remove"), []byte("1\n"), 0200)
+					if err != nil {
+						logger.Debugf("Failed to remove vgpu %q", mdevUUID)
```

Comment:
Probably should be an error log rather than debug. Also should use d.logger to get device and instance context